.gitlab-ci.yml

```diff
@@ -88,24 +111,77 @@ snitch-cluster-vlt:
   script:
     - cd target/snitch_cluster
     - $VERILATOR make bin/snitch_cluster.vlt
-    - $VERILATOR ./sw/test/run.py sw/tests/passing-apps.list --simulator verilator
-    - $VERILATOR ./sw/apps/run.py sw/apps/passing-apps.list --simulator verilator
+    - $VERILATOR ./sw/tests/run.py sw/tests/passing-apps.list --simulator verilator 2>&1 | tee tests.log
+    - $VERILATOR ./sw/apps/run.py sw/apps/passing-apps.list --simulator verilator 2>&1 | tee apps.log
+    - grep -zoP 'All tests passed' tests.log
+    - grep -zoP 'All tests passed' apps.log
   # yamllint enable rule:line-length
+  artifacts:
+    when: on_failure
+    paths:
+      - target/snitch_cluster/tests.log
+      - target/snitch_cluster/apps.log
+    expire_in: 1 day
 
 # VCS
 snitch-cluster-vcs:
   needs: [snitch-cluster-sw]
+  # yamllint disable rule:line-length
   script:
     - cd target/snitch_cluster
     - $VCS make bin/snitch_cluster.vcs
-    - $VCS ./sw/test/run.py sw/tests/passing-apps.list --simulator vcs
-    - $VCS ./sw/apps/run.py sw/apps/passing-apps.list --simulator vcs
+    - $VCS ./sw/tests/run.py sw/tests/passing-apps.list --simulator vcs 2>&1 | tee tests.log
+    - $VCS ./sw/apps/run.py sw/apps/passing-apps.list --simulator vcs 2>&1 | tee apps.log
+    - grep -zoP 'All tests passed' tests.log
+    - grep -zoP 'All tests passed' apps.log
+  # yamllint enable rule:line-length
+  artifacts:
+    when: on_failure
+    paths:
+      - target/snitch_cluster/tests.log
+      - target/snitch_cluster/apps.log
+    expire_in: 1 day
 
 # Questa
 snitch-cluster-vsim:
   needs: [snitch-cluster-sw]
+  # yamllint disable rule:line-length
   script:
     - cd target/snitch_cluster
     - $QUESTA make bin/snitch_cluster.vsim
-    - $QUESTA ./sw/test/run.py sw/tests/passing-apps.list --simulator vsim
-    - $QUESTA ./sw/apps/run.py sw/apps/passing-apps.list --simulator vsim
+    - $QUESTA ./sw/tests/run.py sw/tests/passing-apps.list --simulator vsim 2>&1 | tee tests.log
+    - $QUESTA ./sw/apps/run.py sw/apps/passing-apps.list --simulator vsim 2>&1 | tee apps.log
+    - grep -zoP 'All tests passed' tests.log
+    - grep -zoP 'All tests passed' apps.log
+  # yamllint enable rule:line-length
+  artifacts:
+    when: on_failure
+    paths:
+      - target/snitch_cluster/tests.log
```

Comment:
This output is on the CI console. Why would we also want to artifact it?

Reply:
Good point, I think I added it before I piped the output to the log file and the CI shell simultaneously. Let's remove it.